File: PL.Angular/Models/LoginModel.cs
using System.ComponentModel.DataAnnotations;

namespace PL.Angular.Models
{
    public class LoginModel
    {
        public required string Id { set; get; }
        
        [Required(ErrorMessage = "Email is not specified")]
        public required string Email { get; set; }
         
        [Required(ErrorMessage = "Password is not specified")]
        [DataType(DataType.Password)]
        public required string PasswordCache { get; set; }

        [Required(ErrorMessage = "User role is not specified")]
        public required string UserRole { get; set; }
    }
}

File: PL.Angular/ClientApp/src/app/models/loginModel.ts
export class LoginModel {

    constructor(
        public email: string = '',
        public passwordCache: string = '',
        public id: string = "00000000-0000-0000-0000-000000000000",
        public userRole: string = "user",
    ) {  }

    isValid(): boolean {
        return !!this.email && !!this.passwordCache; 
      }
}